touchforms/formplayer/static/formplayer/script/xforgasm.js

```diff
@@ -16,8 +16,52 @@ var ERROR_MESSAGE = "Something unexpected went wrong on that request. " +
     "If you have problems filling in the rest of your form please submit an issue. " +
     "Technical Details: ";
 
+function TaskQueue() {
+    this.queue = [];
+};
+
+/*
+ * Executes the queue in a FIFO action. If name is supplied, will execute the first
+ * task for that name.
+ */
+TaskQueue.prototype.execute = function(name) {
+    var task,
+        idx;
+    if (name) {
+        idx = _.indexOf(_.pluck(this.queue, 'name'), name);
+        task = this.queue.slice(idx, idx + 1)[0];
+        this.queue = [].concat(this.queue.slice(0, idx), this.queue.slice(idx + 1));
```

Comment:
I think this would be more efficiently (and more correctly if a task with the given name is not found) done like this:
```js
idx = _.findIndex(this.queue, function (task) {
    return task.name === name;
});
if (idx !== -1) {
    task = this.queue.splice(idx, 1)[0];
}
```